subprojects/core/src/test/groovy/org/gradle/api/internal/changedetection/changes/DefaultTaskArtifactStateRepositoryTest.groovy

```diff
@@ -716,29 +717,25 @@ class DefaultTaskArtifactStateRepositoryTest extends AbstractProjectBuilderSpec
         fileSystemMirror.beforeTaskOutputChanged()
     }
 
+    @EqualsAndHashCode
```

Comment:
What advantage does the `EqualsAndHashCode` implementation give here? Are we using the class in a collection?

Reply:
None, removed. `ToString` is the important one here.